packages/vite/src/node/utils.ts

```diff
@@ -502,8 +503,7 @@ export function generateCodeFrame(
 
 export function isFileReadable(filename: string): boolean {
   try {
-    // The "throwIfNoEntry" is a performance optimization for cases where the file does not exist
-    if (!fs.statSync(filename, { throwIfNoEntry: false })) {
+    if (!tryStatSync(filename)) {
       return false
     }
```

Comment:
LGTM! Can we move this if condition outside of the try block? Since `tryStatSync` shouldn't trigger any error now.